src/libs/antares/study/binding_constraint/BindingConstraintGroup.cpp

```diff
@@ -0,0 +1,39 @@
+//
+// Created by marechaljas on 28/06/23.
+//
+
+#include "BindingConstraintGroup.h"
+
+#include <algorithm>
+#include <utility>
+
+namespace Antares::Data {
+
+    void BindingConstraintGroup::add(const std::shared_ptr<BindingConstraint> &constraint) {
+        constraints_.insert(constraint);
+    }
+
+    BindingConstraintGroup::BindingConstraintGroup(std::string name) :
+            name_(std::move(name)) {
+
+    }
+
+    std::set<std::shared_ptr<BindingConstraint>> BindingConstraintGroup::constraints() const {
+        return constraints_;
+    }
+
+    void BindingConstraintGroup::fixTSNumbersWhenWidthIsOne() {
+        if (std::all_of(constraints_.begin(), constraints_.end(), [](auto constraint){
+            return constraint->RHSTimeSeries().width == 1;
+        })) {
+            timeSeriesNumbers_.timeseriesNumbers.fillColumn(0, 0);
+        }
+    }
+
+    unsigned BindingConstraintGroup::numberOfTimeseries() const {
+        //Assume all BC in a group have the same width
+        if (constraints_.empty()) return 0;
+        return (*constraints_.begin())->RHSTimeSeries().width;
```

Comment:
I'm not sure to understand properly, I'm not null checking
`if (constraints_.empty()) return 0;` => If no constraint exist for this group, number of TS is 0. Should not happen in practice but happen in unit tests.

`return (*constraints_.begin())->RHSTimeSeries().width;` 
=> `constraints_.begin()` is iterator. Always valid since from the previous check we know the set is not empty.
=> `(*constraints_.begin())` is the shared_ptr to the constraint